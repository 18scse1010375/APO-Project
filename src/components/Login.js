import React from "react";
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import LoginStyle from './LoginStyle.css';
import { useState, useEffect } from "react";
import { Navigate } from "react-router-dom";
import { adobeIms } from "../services/AdobeIms.ts";
import { userStore } from "store/UserStore";
import { Button, defaultTheme, Provider } from '@adobe/react-spectrum';

const Login = () => {
  const [isLogin, setIsLogin] = useState(false);
  useEffect(() => {
    if (adobeIms.isSignedInUser()) {

      setIsLogin(true)
    }

  }, [userStore.isImsInit]);

  const signIn = () => {
    adobeIms.signIn();
  }
  return (

    isLogin ? (
      <Navigate to="/home" replace={true} />
    ) : <>

      <div id="login-page" className="text-center" >
        <p>Welcome to APO Manager Console</p>
        <div id="login-button">
          <Provider theme={defaultTheme}>
            <Button variant="accent" onClick={signIn} > Login </Button>
          </Provider>
        </div>
      </div>
    </>
  )
}
export default Login;
