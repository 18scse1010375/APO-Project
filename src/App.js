// import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import MakeHeader from './components/MakeHeader';
import { BrowserRouter as Router, Routes, Route,useLocation} from 'react-router-dom'
import MakeCard from './components/MakeCard';
import MakeLabel from './components/MakeLabel';
import Login from './components/Login';
import { useEffect, useState } from 'react';
import BulkEmail from './components/BulkEmail';
import 'react-toastify/dist/ReactToastify.css';
import Geeks from 'components/Geeks';
import { NoMatch } from './components/NoMatch';

import { setEnv } from 'configs';
import { adobeIms } from './services/AdobeIms';
import { userStore } from './store/UserStore';


export const LocationDisplay = () => {
  const location = useLocation()

  return <div data-testid="location-display">{location.pathname}</div>
}











function App() {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [isLogin, setIsLogin] = useState(false);
  useEffect(() => {
    if (!userStore.isImsInit) {
      adobeIms.initialize();
      userStore.setImsInit(true);
    }

    if (adobeIms.isSignedInUser()) {
      setIsLogin(true)
    }

  }, [userStore.isImsInit]);

  useEffect(() => {
    fetch("/env")
      .then(async res => setEnv(await res.json()));
  })
  return (
    <>
    
      {/* <Router> */}
        <Routes>
          <Route exact path='/home' element={< MakeCard />}></Route>
          <Route exact path='/email-suppresion' element={< MakeLabel />}>  </Route>
          <Route exact path='/bulk-email-sent' element={< BulkEmail />}>  </Route>
          <Route exact path='/' element={<Login />} >  </Route>
          <Route exact path='/logout' element={<Login />} >  </Route> 
          <Route path="*" element={<NoMatch />} />

        </Routes>
      {/* </Router> */}

            <LocationDisplay/>

    </>
  );
}

export default App;
