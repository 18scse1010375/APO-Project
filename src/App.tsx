import { Header } from 'Header';
import './App.css';
import { adobeIms, login, updateUserProfile} from './services/AdobeIms';
import { userStore } from './store/UserStore';
import {useEffect} from 'react';

const signIn = ()=> {
  adobeIms.signIn();
}

const logout = () => {
  try {
    adobeIms.signOut();
    let profile = null;
    userStore.updateData(profile, 'token');
  } catch (e) {
    console.error('logout', e);
  }
}

function App() {

  useEffect(() => {
    if (!userStore.isImsInit) {
      adobeIms.initialize();
      userStore.setImsInit(true);
    }
  }, [userStore.isImsInit]);

  return (
    <div className="App">
      <Header store = {userStore}></Header>
      <header className="App-header">
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <button onClick={signIn}>Sign in</button>
        <button onClick={logout}>Sign out</button>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
