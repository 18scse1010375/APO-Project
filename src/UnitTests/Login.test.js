/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable no-undef */
import { render, screen , fireEvent } from '@testing-library/react';
// import Demo from './Demo';
import { shallow } from "enzyme";
import React from 'react';
import App from 'App';

import { Router, Route } from "react-router";
import { createMemoryHistory } from 'history';



import '@testing-library/jest-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


import Login from 'components/Login';
import { BrowserRouter } from 'react-router-dom';
configure({ adapter: new Adapter() });



describe("<Login/>", () => {
  let component

    // test("check the default component url path" , ()=>{
    //   const history = createMemoryHistory();
    //   render(
    //     <Router history={history}>
    //       <App signedInUser={null} />
    //     </Router>
        
    //   );
    //   // expect(history.location.pathname).toBe("/login");


    // }  )

    // beforeEach(() => {
    //    component = shallow(< Login/>)
    // })
  
    // test('is initially active: false', () => {
    //   component.state('isLogin').should.be.false
    //   // component.find('div.fade-in').prop('className').should.equal('fade-in')
    // })






    test('render the title of an application', () => {
      render(<Login/>);
   
      const titleEl = screen.getByText(/Welcome to APO Manager Console/);
      expect(titleEl).toBeInTheDocument();
    });

    test('render buttons', () => {
        render(<Login/>);
     
        const button1 = screen.getByTestId("login");

     
        expect(button1).toBeInTheDocument();   //verification for the wheather login button is present or not
        expect(button1.textContent).toEqual(" Login ");
        expect(button1.getAttribute("disabled")).toBe(null)    //To check login button is enabled or not
 
        // fireEvent.click(button1)  //verification to check that login button is clcikable or not


        
       
    
      });


      test("Taking Snapshot of Login Component", ()=>{
        const wrapper=shallow( <Login/> )
        expect(wrapper).toMatchSnapshot();
        
        



      } )

})