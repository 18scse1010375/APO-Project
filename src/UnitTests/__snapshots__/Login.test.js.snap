// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Taking Snapshot of Login Component 1`] = `
<Fragment>
  <div
    className="text-center"
    id="login-page"
  >
    <h2>
      Welcome to APO Manager Console
    </h2>
    <div
      id="login-button"
    >
      <ForwardRef($96cabdf5b0aed55b$var$Provider)
        theme={
          Object {
            "dark": Object {
              "spectrum--darkest": "spectrum--darkest_256eeb",
            },
            "global": Object {
              "spectrum": "spectrum_9e130c",
              "spectrum--dark": "spectrum--dark_9e130c",
              "spectrum--darkest": "spectrum--darkest_9e130c",
              "spectrum--large": "spectrum--large_9e130c",
              "spectrum--light": "spectrum--light_9e130c",
              "spectrum--lightest": "spectrum--lightest_9e130c",
              "spectrum--medium": "spectrum--medium_9e130c",
            },
            "large": Object {
              "spectrum--large": "spectrum--large_c40598",
            },
            "light": Object {
              "spectrum--light": "spectrum--light_a40724",
            },
            "medium": Object {
              "spectrum--medium": "spectrum--medium_4b172c",
            },
          }
        }
      >
        <ForwardRef($842a195ed1671b15$var$Button)
          data-testid="login"
          onClick={[Function]}
          variant="accent"
        >
           Login 
        </ForwardRef($842a195ed1671b15$var$Button)>
      </ForwardRef($96cabdf5b0aed55b$var$Provider)>
    </div>
  </div>
</Fragment>
`;
